src/main/java/recipeapplication/database/UserFirebaseCrud.java

```diff
@@ -1,49 +1,108 @@
 package recipeapplication.database;
 
+import Utils.FinalsStringsExceptions;
+import Utils.StringUtils;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.cloud.firestore.*;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.HttpStatus;
 import org.springframework.stereotype.Component;
 
 import com.google.cloud.firestore.Firestore;
 
+import org.springframework.web.server.ResponseStatusException;
 import recipeapplication.components.User;
 
+import java.util.Map;
+
 @Component
 public class UserFirebaseCrud implements FirebaseCrud<User> {
-	
-	private Firestore firestore;
-	private String collection;
-	
-	@Autowired
-	UserFirebaseCrud(Firestore firestore, 
-			@Value("${recipeapp.firestore.collection.users:users}") String collection) {
-		this.firestore = firestore;
-		this.collection = collection;
-	}
-
-
-	@Override
-	public void create(User user) {
-		this.firestore.collection(collection).document(user.getId()).set(user);
-		
-	}
-
-	@Override
-	public void read(String document) {
-		// TODO Auto-generated method stub
-		
-	}
-
-	@Override
-	public void update(User user) {
-		// TODO Auto-generated method stub
-		
-	}
-
-	@Override
-	public void delete(String document) {
-		// TODO Auto-generated method stub
-		
-	}
+
+    private Firestore firestore;
+    private String collection;
+    private ObjectMapper mapper; // jackson's objectmapper
+
+    @Autowired
+    UserFirebaseCrud(Firestore firestore,
+                     @Value("${recipeapp.firestore.collection.users:users}") String collection) {
+        this.firestore = firestore;
+        this.collection = collection;
+        this.mapper = new ObjectMapper();
+    }
+
+    @Override
+    public User create(User user) throws Exception {
+        if (!isAllUserDetailsValid(user)) {
```

Comment:
You are using @Valid annotation alongside @NotNull annotations, user should not contain null values at this point (but please double check this)